stream-parent/stream-core/src/main/java/org/interledger/stream/StreamPacket.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * <p>When a connector sees an incoming prepared transfer with an ILP Payment, the receiver reads
+ * the ILP Payment to confirm the details of the packet. For example, the connector reads the InterledgerAddress of the
+ * payment's receiver, and if the connector has a route to the receiver's account, the connector prepares a transfer to
+ * continue the payment, and attaches the same ILP Payment to the new transfer. Likewise, the receiver confirms that the
+ * amount of the ILP Payment Packet matches the amount actually delivered by the transfer. And finally, the receiver
+ * decodes the data portion of the Payment and matches the condition to the payment.</p>
+ *
+ * <p>The receiver MUST confirm the integrity of the ILP Payment, for example with a hash-based
+ * message authentication code (HMAC). If the receiver finds the transfer acceptable, the receiver releases the
+ * fulfillment for the transfer, which can be used to execute all prepared transfers that were established prior to the
+ * receiver accepting the payment.</p>
+ */
+public interface StreamPacket {
+
+  int VERSION_1 = 1;
+
+  /**
+   * Get the default builder.
+   *
+   * @return a {@link StreamPacketBuilder} instance.
+   */
+  static StreamPacketBuilder builder() {
+    return new StreamPacketBuilder();
+  }
+
+  /**
+   * The version of this packet, currently 1.
+   *
+   * @return This packet's STREAM version (always 1).
+   */
+  default short version() {
+    return VERSION_1;
+  }
+
+  /**
+   * ILPv4 packet type this STREAM packet MUST be sent in (12 for Prepare, 13 for Fulfill, and 14 for Reject).
+   *
+   * @return A {@link InterledgerPacketType}
+   */
+  InterledgerPacketType interledgerPacketType();
+
+  /**
+   * Identifier for an ILP request / response. Clients and Servers track their own outgoing packet sequence numbers and
+   * increment the Sequence for each ILP Prepare they send. The Receiver MUST respond with a STREAM packet that includes
+   * the same Sequence as the Sender's Prepare packet. A sender MUST discard a STREAM packet in which the Sequence does
+   * not match the STREAM packet sent with their ILP Prepare.
+   *
+   * @return A {@link UnsignedLong} for this packet's sequence.
+   */
+  UnsignedLong sequence();
+
+  /**
+   * If the STREAM packet is sent on an ILP Prepare, this represents the minimum the receiver should accept (this is a
+   * signal between the sender and receiver that cannot be tampered with by intermediate nodes, and acts as a safety net
+   * against fees that might be too high in the newtwork). If the packet is sent on an ILP Fulfill or Reject, this
+   * represents the amount that the receiver got in the Prepare.
+   *
+   * @return A {@link UnsignedLong} for this packet's prepareAmount.
+   */
+  UnsignedLong prepareAmount();
+
+  /**
+   * A list of {@link StreamFrameType}.
+   *
+   * @return A {@link List} of Stream Frames that this packet contains.
+   */
+  List<StreamFrame> frames();
+
+  @Immutable
```

Comment:
Same question here. Abstract class seems unnecessary unless StreamPacket gets implemented in other places which I didn't see.

Reply:
See answer above.